app/controllers/cards_controller.rb

```diff
@@ -31,6 +32,8 @@ def create
     # Set the new card as the last card of the list
     @card.previous_id = @list.last_card.try(:id)
 
+    @card.description = dradify_form if params[:item_form]
```

Comment:
I don't like how multiple controllers are peppered for this addition. Is it something we could add into the `FormDradifier` module? Something like:

disclaimer: this is untested
```ruby
before_filter :set_content

def set_content
  return unless params[:item_form]

  resource = @controller.instance_variable_get("@#{instance_name}")

  if resource&.description
    'description'
  elsif resource&.content
    'content'
  elsif resource&.text
    'text'
  end

  resource.send("#{method}=", dradify_form)
end

Reply:
@brianp  I tested this idea and noticed that it'll only work if we declare the `include` after we declare the `before_action` callbacks in the controller:
```
before_action ....
include FormDradifier
```

This is because we use the item instance (`@issue`, `@note`, etc.) in`set_content` but they only get defined in the other `before_action` callback. Moving the `FormDradifier` inclusion after the callbacks also moves the `set_content` call, and only then we'll have the item instance available for us.

Note that the same thing happens if we use `around_action` because rails pushes the callbacks sequentially: [callbacks.rb](https://github.com/rails/rails/blob/157920aead96865e3135f496c09ace607d5620dc/actionpack/lib/abstract_controller/callbacks.rb#L91)

I do like the improvement of moving this code to the module. It looks a bit cleaner but it's also inconsistent with what we have. Thoughts?

Reply:
We should add a class method into `FormDradifier` that allows us to include it into controllers at the top as normal and then we can call something like `dradify_form only: %w[create update]`

The method `dradify_form` could do one of two things

- Define the before filter for the method behind the scenes.
- Call the `set_content` method directly assuming it has been placed under other before filters.

Alternatively (and simplest): Leave the logic in the module, and move the `before_filter` from the module to the controller. Allowing the controller to decide the flow. It still requires an addition in the controllers, but I feel it's cleaner than additions per action.

Reply:
Thanks for the suggestions @brianp . I went ahead and implemented the `before_action` in the controllers. I encountered some nuances in the cards controller and the issues' merge controller but overall the implementation looks much cleaner.